zed_test.go

```diff
@@ -143,7 +144,7 @@ func runOneBoomerang(t *testing.T, format, data string) {
 		ast, err := parser.ParseQuery("fuse")
 		require.NoError(t, err)
 		rctx := runtime.NewContext(context.Background(), zctx)
-		q, err := compiler.NewCompiler().NewQuery(rctx, ast, []zio.Reader{dataReadCloser})
+		q, err := compiler.NewCompiler(storage.NewLocalEngine()).NewQuery(rctx, ast, []zio.Reader{dataReadCloser}, 0)
```

Comment:
Nit: This path doesn't use the engine.
```suggestion
		q, err := compiler.NewCompiler(nil).NewQuery(rctx, ast, []zio.Reader{dataReadCloser}, 0)
```